contrib/gitian-builder/gitian-build.sh

```diff
@@ -0,0 +1,428 @@
+      proc=$2
+      shift
+    else
+      echo 'Error: "-j" requires an argument'
+      exit 1
+    fi
+    ;;
+  # Memory to allocate
+  -m)
+    if [ -n "$2" ]; then
+      mem=$2
+      shift
+    else
+      echo 'Error: "-m" requires an argument'
+      exit 1
+    fi
+    ;;
+  # URL
+  -u)
+    if [ -n "$2" ]; then
+      url=$2
+      shift
+    else
+      echo 'Error: "-u" requires an argument'
+      exit 1
+    fi
+    ;;
+  # Detach sign
+  --detach-sign)
+    commitFiles=false
+    ;;
+  # Commit files
+  --no-commit)
+    commitFiles=false
+    ;;
+  # Setup
+  --setup)
+    setup=true
+    ;;
+  --test)
+    test=true
+    ;;
+  *) # Default case: If no more options then break out of the loop.
+    break ;;
+  esac
+  shift
+done
+
+echo "Using ${proc} CPU and ${mem} RAM"
+
+
+if [[ $test == true ]]; then
+  if [[ $commit == true ]]; then
+    VERSION="f80bfe9068ac1a0619d48dad0d268894d926941e"
+  else
+    VERSION="1.14.3"
+  fi
+  DESCRIPTORS=('test')
+  SIGN_DESCRIPTORS=()
+  COMMIT=$VERSION
+  SIGNER="signer"
+  SHA256SUM="0d519f6ade0e601617a7c44b764eeae35a8784070c3e44f542011956f1743459"
+fi
+
+# Get signer
+if [[ -n "$1" ]]; then
+  SIGNER=$1
+  shift
+fi
+
+# Get version
+if [[ -n "$1" ]]; then
+  VERSION=$1
+  COMMIT=$VERSION
+  shift
+fi
+
+# Add a "v" if no -c
+if [[ $commit == false ]]; then
+  COMMIT="v${VERSION}"
+fi
+
+#############################
+######## ENV TESTING ########
+#############################
+if [[ $setup == true || $build == true ]]; then
+  echo "Testing Docker..."
+  if ! docker ps &>/dev/null; then
+    echo "Docker is not launched..."
+    exit 1
+  fi
+fi
+
+if [[ $sign == true ]]; then
+  echo "Testing GPG Keys available..."
+  result=$(gpg --list-secret-keys --keyid-format=long | grep sec | grep -v revoked | grep "" -c)
+  if [[ $result == "" ]]; then
+    echo "No GPG keys available..."
+    echo "Please follow this documentation: https://docs.github.com/en/github/authenticating-to-github/managing-commit-signature-verification/generating-a-new-gpg-key"
+    exit 1
+  fi
+
+  # Check that a signer is specified
+  if [[ $SIGNER == "" ]]; then
+    echo "$scriptName: Missing signer."
+    echo "Try $scriptName --help for more information"
+    exit 1
+  fi
+fi
+
+if [[ $push == false && $setup == false ]]; then
+  # Check that a version is specified
+  if [[ $VERSION == "" ]]; then
+    echo "$scriptName: Missing version."
+    echo "Try $scriptName --help for more information"
+    exit 1
+  fi
+fi
+
+#######################
+######## SETUP ########
+#######################
+if [[ $setup == true ]]; then
+  git clone https://github.com/dogecoin/gitian.sigs.git
+  git clone https://github.com/dogecoin/dogecoin-detached-sigs.git
+  git clone https://github.com/devrandom/gitian-builder
+
+  pushd gitian-builder || exit 1
+
+  git fetch origin 9e97a4d5038cd61215f5243a37c06fa1734a276e # LAST VERSION TESTED
+  git reset --hard FETCH_HEAD
+  git am < "$dirName"/patches/0001-Docker-apt-cacher.patch
```

Comment:
Using a fresh install, patch fail if git email is not configured using something like `git config --global user.email "you@example.com"`.